compiler/rustc_typeck/src/coherence/inherent_impls.rs

```diff
@@ -387,7 +412,34 @@ impl InherentCollect<'tcx> {
                     lang,
                     ty
                 )
-                .span_help(span, "consider using a trait to implement these methods")
+                .span_help(
```

Comment:
I know you didn't change this, but I think this should be `.help`, not `.span_help`. Using `span_help` causes it to reprint the source code, which is not helpful and also confusing because it's exactly the same.

```suggestion
                .help(
```